cli/src/cli_util.rs

```diff
@@ -497,36 +497,36 @@ impl ReadonlyUserRepo {
     }
 }
 
-/// A branch that should be advanced to satisfy the "advance-branches" feature.
-/// This is a helper for `WorkspaceCommandTransaction`. It provides a type-safe
-/// way to separate the work of checking whether a branch can be advanced and
-/// actually advancing it. Advancing the branch never fails, but can't be done
-/// until the new `CommitId` is available. Splitting the work in this way also
-/// allows us to identify eligible branches without actually moving them and
-/// return config errors to the user early.
+/// A bookmark that should be advanced to satisfy the "advance-bookmarkes"
+/// feature. This is a helper for `WorkspaceCommandTransaction`. It provides a
+/// type-safe way to separate the work of checking whether a bookmark can be
+/// advanced and actually advancing it. Advancing the bookmark never fails, but
+/// can't be done until the new `CommitId` is available. Splitting the work in
+/// this way also allows us to identify eligible bookmarkes without actually
+/// moving them and return config errors to the user early.
 pub struct AdvanceableBranch {
     name: String,
     old_commit_id: CommitId,
 }
 
-/// Helper for parsing and evaluating settings for the advance-branches feature.
-/// Settings are configured in the jj config.toml as lists of [`StringPattern`]s
-/// for enabled and disabled branches. Example:
+/// Helper for parsing and evaluating settings for the advance-bookmarkes
+/// feature. Settings are configured in the jj config.toml as lists of
+/// [`StringPattern`]s for enabled and disabled bookmarkes. Example:
 /// ```toml
-/// [experimental-advance-branches]
-/// # Enable the feature for all branches except "main".
-/// enabled-branches = ["glob:*"]
-/// disabled-branches = ["main"]
+/// [experimental-advance-bookmarkes]
+/// # Enable the feature for all bookmarkes except "main".
+/// enabled-bookmarkes = ["glob:*"]
+/// disabled-bookmarkes = ["main"]
 /// ```
 struct AdvanceBranchesSettings {
-    enabled_branches: Vec<StringPattern>,
-    disabled_branches: Vec<StringPattern>,
+    enabled_bookmarkes: Vec<StringPattern>,
+    disabled_bookmarkes: Vec<StringPattern>,
 }
 
 impl AdvanceBranchesSettings {
     fn from_config(config: &config::Config) -> Result<Self, CommandError> {
         let get_setting = |setting_key| {
-            let setting = format!("experimental-advance-branches.{setting_key}");
+            let setting = format!("experimental-advance-bookmarkes.{setting_key}");
```

Comment:
I guess this should be reverted

Reply:
I remember there was talk about keeping this flag the same, but maybe we should accept both here as well? (preferring `experimental-advance-bookmarks` if it's set but also checking for the old config)

Reply:
We surely could accept both but config flag deprecations are quite a bit harder than command deprecations. It also is super churny for users. I just said that to not break everything even more :-).

Reply:
reverted, thanks for finding it.